### ISSUE TITLE
Drift using 2D Laser in large spaces

### ISSUE THREAD

**Issue by LotfiZ** [vectr-ucla/direct_lidar_odometry#44]
Hi dear,

its been few weeks that we are testing DLO to compute odometry for a mobile robot and its has great result. However, when we went to a larger space to test it, we had a lot of drift, we did some parameter tuning as suggested in the previous issue tickets concerning sparse lidars but we could not reach acceptable results. The laser measurements are in 2D and do not cover full 360 but around 270, in addition, sometimes the laser measures the ground. Do you think that its related  to this ?  

![image](https://github.com/vectr-ucla/direct_lidar_odometry/assets/41443752/e424bea8-7fb7-41c9-bb70-612bfbe09beb)

We played with the outlier rejection parameters, the max corr distance for both s2s and s2m and the adaptiveParams. An error is raised everytime in this environnement and its related to convexHull : `[pcl::ConvexHull::performReconstrution3D] ERROR: qhull was unable to compute a convex hull for the given point cloud (375)!`. This error does not rise in smaller places.

Thank you very much for your help !

**Comment by kennyjchen** [comment]
Yes, a non-360 FOV can definitely affect how submapping is done in DLO.

**Comment by aba01kz** [comment]
sorry, can I use DLO with rplidar (2D lidar with single line laser)?

**Comment by kennyjchen** [comment]
You probably could use it, but I'd imagine the performance will be pretty poor unfortunately. DLO is optimized for 3D point clouds with 360 FOV.
